src/crh/mod.rs

```diff
@@ -14,12 +14,29 @@ pub mod constraints;
 #[cfg(feature = "r1cs")]
 pub use constraints::*;
 
-pub trait FixedLengthCRH {
+pub trait CRH {
     const INPUT_SIZE_BITS: usize;
 
     type Output: ToBytes + Clone + Eq + core::fmt::Debug + Hash + Default;
     type Parameters: Clone + Default;
 
-    fn setup<R: Rng>(r: &mut R) -> Result<Self::Parameters, Error>;
+    fn setup_crh<R: Rng>(r: &mut R) -> Result<Self::Parameters, Error>;
     fn evaluate(parameters: &Self::Parameters, input: &[u8]) -> Result<Self::Output, Error>;
 }
+
+pub trait TwoToOneCRH {
+    /// The size for left input and right input
+    const LEFT_INPUT_SIZE_BITS: usize;
+    const RIGHT_INPUT_SIZE_BITS: usize;
+
+    type Output: ToBytes + Clone + Eq + core::fmt::Debug + Hash + Default;
+    type Parameters: Clone + Default;
+
+    fn setup_two_to_one_crh<R: Rng>(r: &mut R) -> Result<Self::Parameters, Error>;
+    fn evaluate_two_to_one_hash(
```

Comment:
Same, I presume in generic contexts it'll be called like `TwoToOneCRH::evaluate`

Reply:
If a CRH implements both CRH and TwoToOneCRH, user might need to use fully qualified syntax to call this method... 

Anyway, seems that the renaming will have more drawbacks than its benefit. I will revert that rename shortly. Sorry for confusion

Reply:
Hmm in generic contexts, it's likely that we'll have type parameters like `H1: CRH, H2: TwoToOneCRH`, which avoids the ambiguity. Not sure what to do here, what do you think @ValarDragon ?

Reply:
resolved. For now, in non-generic context: 
to use `CRH`, call `<pedersen::CRH as CRH>::evaluate()`. 
to use `TwoToOneCRH`, call `<pedersen::CRH as TwoToOneCRH>::evaluate()`